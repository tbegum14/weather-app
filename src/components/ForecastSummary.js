import React from "react";
import iconData from '../data/iconData.json'


export default function ForecastSummary(props){
    const {date, temperature, description, icon, onSelect } = props
    const weatherCode = `${icon.slice(0,1)}00`;
    const formattedDate= new Date(date).toDateString()
    return <div className="forecast-summary" data-testid="forecast-summary">
        <h3 className="forecast-summary_date">{formattedDate}</h3>
        <div className="forecast-summary_icon" data-testid="forecast-icon" >
            <img src={iconData[weatherCode]} alt="icon" />
        </div>
        <h2 className="forecast-summary_temperature">{temperature}<span>&#8451;</span></h2>
        <h2 className="forecast-summary_description">{description}</h2>
        <button type="button" onClick={()=>onSelect(date)}>More details</button>
    </div>
}