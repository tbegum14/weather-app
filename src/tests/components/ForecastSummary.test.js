import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("Forecast Summary", () => {
	const validProps = {
		date: 1111111,
		description: "stubdescription",
		icon: "stubIcon",
		temperature: {
			min: 1,
			max: 10,
		},
		onSelect: ()=>{}
	};

	it("displays the correct values", () => {
		const { asFragment } = render(
			<ForecastSummary
				date={validProps.date}
				description={validProps.description}
				icon={validProps.icon}
				temperature={validProps.temperature.max}
			/>
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it("renders with the correct values", () => {
		const { getByText, getByTestId } = render(
			<ForecastSummary
				date={validProps.date}
				description={validProps.description}
				icon={validProps.icon}
				temperature={validProps.temperature.max}
				onSelect={validProps.onSelect}
			/>
		);
        expect(getByText("Thu Jan 01 1970")).toHaveAttribute("class", "forecast-summary_date")
		expect(getByText("stubdescription")).toHaveAttribute(
			"class",
			"forecast-summary_description"
		);
		expect(getByTestId("forecast-icon")).toHaveAttribute(
			"class",
			"forecast-summary_icon"
		);
		expect(getByText("10")).toHaveAttribute(
			"class",
			"forecast-summary_temperature"
		);
	});
});
