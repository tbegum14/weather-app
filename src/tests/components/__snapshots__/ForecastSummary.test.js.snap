// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Summary displays the correct values 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <h3
      class="forecast-summary_date"
    >
      Thu Jan 01 1970
    </h3>
    <div
      class="forecast-summary_icon"
      data-testid="forecast-icon"
    >
      <img
        alt="icon"
      />
    </div>
    <h2
      class="forecast-summary_temperature"
    >
      10
      <span>
        â„ƒ
      </span>
    </h2>
    <h2
      class="forecast-summary_description"
    >
      stubdescription
    </h2>
    <button
      type="button"
    >
      More details
    </button>
  </div>
</DocumentFragment>
`;
